source/solutions-library/payments-solution.txt

```diff
@@ -0,0 +1,401 @@
+       "balance": Number, 
+       "limitations": {
+           "withdrawalLimit": Number, 
+           "transferLimit": Number, 
+           "otherLimitations": String
+       }, 
+       "securityTags": [String], 
+       "encryptedDetails": String
+   }
+
+3. **Transactions collection** 
+
+.. code-block:: json
+
+   {
+       "_id": ObjectId, 
+       "account_id": ObjectId, 
+       "amount": Number, 
+       "date": Date, 
+       "type": String, // e.g., 'debit', 'credit' 
+       "status": String, // e.g., 'completed', 'pending', 'refund' 
+       "details": String, // Encrypted if sensitive 
+       "referenceData": {
+           "sender": { ... }, 
+           "receiver": { ... }, 
+           "steps": [{ ... }],
+           "relatedTransactions": [{ ... }], 
+           "notes": String, 
+           "reportingTags": [String]
+       }
+   }
+
+4. **Notifications collection** 
+
+.. code-block:: json
+
+   {
+       "_id": ObjectId, 
+       "relatedEntityId": ObjectId, 
+       "userIds": [ObjectId], 
+       "message": String, 
+       "createdAt": Date, 
+       "statuses": [
+           {
+               "userId": ObjectId, 
+               "status": String // e.g., 'unread', 'read'
+           }
+       ]
+   }
+
+Building the Solution
+---------------------
+The code for the following step-by-step guide can be found in this `Github repo
+<https://github.com/mongodb-developer/FSIDemo-Payments/tree/main>`__.
+
+Features Covered
+----------------
+**Cross microservices:** 
+
+- Indexing and scalability 
+
+- |json| schema validation 
+
+- Permission and data segregation 
+
+- Auditing
+
+**User and account microservices:** 
+
+- `Document model <https://www.mongodb.com/document-databases>`__: Flexible user and
+  account structure for different accounts and user profiles 
+
+- `Kafka streaming sink
+  <https://www.mongodb.com/docs/kafka-connector/current/sink-connector/configuration-properties/>`__:
+  Data being streamed from external sources 
+
+- Transactions (User to Account references): Keeping account and user data `ACID-compliant
+  <https://www.mongodb.com/products/capabilities/transactions>`__ 
+
+- In-use `encryption
+  <https://www.mongodb.com/products/capabilities/security/encryption>`__ 
+
+- Full-text search: Account IDs and usernames are searchable for users to pick via
+  `full-text search <https://www.mongodb.com/atlas/search>`__.
+
+**Transactions and payments microservices:** 
+
+- `Change streams <https://www.mongodb.com/basics/change-streams>`__: Payments are
+  event-driven by transactions 
+
+- `Time series <https://www.mongodb.com/products/capabilities/time-series>`__ for
+  transaction history: Transaction history is built in a time series collection 
+
+- In-use `encryption
+  <https://www.mongodb.com/products/capabilities/security/encryption>`__
+
+**Notification microservices:** 
+
+- {+kafka+} source: Notifications are being down-streamed to external systems and users via {+kafka+} 
```

Comment:
```suggestion
- {+kafka+} source: Notifications are being down-streamed to external systems and users via Kafka 
```
These don't both need to be links, so suggest making at least one of them plaintext. We don't generally expand substitutions to links, it oversaturates the user.